### PR TITLE
feat(model-server): log the cause of internal server errors

### PR DESCRIPTION
**Pull request by odzhychko**
Log the cause of internal server errors.
This should help understand internal server errors better.
It gives users the possibility to a stack trace to bug reports.

## To be verified by reviewers

* [x] Relevant public API members have been documented
* [x] Documentation related to this PR is complete
  * [x] Boundary conditions are documented
  * [x] Exceptions are documented
  * [x] Nullability is documented if used
* [x] Touched existing code has been extended with documentation if missing
* [x] Code is readable
* [ ] New features and fixed bugs are covered by tests

### CODE REVIEW

File: model-server/src/main/kotlin/org/modelix/model/server/Main.kt [hunk 1]
```diff
@@ -310,6 +310,8 @@ object Main {
             }
 
             exception<Throwable> { call, cause ->
+                // Such errors were most likely not caught and logged anywhere else.
+                LOG.error("Encountered internal server error.", cause)
```
Comment (languitar) [inline model-server/src/main/kotlin/org/modelix/model/server/Main.kt:314]:
```suggestion
                LOG.error("Encountered an internal server error.", cause)
```

Comment (odzhychko) [reply model-server/src/main/kotlin/org/modelix/model/server/Main.kt]: thx

**Comment by languitar** [comment]
Do we want to test this somehow? From a user perspective, being able to - for instance - install log alerting with Loki, is a good addition and we should ensure that this doesn't break again accidentally.

**Comment by odzhychko** [comment]
I see the benefit of treating it as a stable feature, but for me, this is out-of-scope for this PR.
Especially, I do not want to decide now the way we want to log such messages or any other details.

As the first step, I just want to log the exception to make debugging of "Internal Server" errors easier. This would have helped with finding #795 faster.

**Review by slisson — APPROVED** [review]

**Review by languitar — APPROVED** [review]
I would have been fine with just asserting that we log something in case of an internal server error. But fine for the moment.

**Pull request merged**